tests/integration/e2e/SchedulePage.test.tsx

```diff
@@ -0,0 +1,199 @@
+            course: {
+              termPattern: TERM_PATTERN.BOTH,
+            },
+          },
+        },
+      });
+      const parentMeetingCourse = parentMeetings[0].courseInstance.course;
+      childMeetings = await meetingRepository.find({
+        relations: ['courseInstance', 'courseInstance.course', 'courseInstance.course.area', 'room', 'room.building'],
+        where: {
+          day: Not(parentMeetings[0].day),
+          courseInstance: {
+            course: {
+              id: Not(parentMeetingCourse.id),
+              area: {
+                name: Not(parentMeetingCourse.area.name),
+              },
+            },
+          },
+        },
+      });
+      renderResult = render(
+        <MemoryRouter initialEntries={['/course-admin']}>
+          <App />
+        </MemoryRouter>
+      );
+    });
+    context('when a course is a child of another', function () {
+      it('should display the meeting information of the parent course', async function () {
+        const testMeeting = parentMeetings[0];
+        const testParentCourse = testMeeting.courseInstance.course;
+        const testChildCourse = childMeetings[0].courseInstance.course;
+        await renderResult.findAllByText(testChildCourse.title);
+
+        // Before setting the child course's "Same As" value to the parent,
+        // we are filtering the Course Admin table by area, because there are
+        // courses with the same title and differing catalog numbers. When
+        // selecting the edit button, we want to make sure we are selecting
+        // the intended one.
+        const areaLabelText = 'The table will be filtered as selected in this area dropdown filter';
+        const areaFilter = renderResult
+          .getByLabelText(areaLabelText) as HTMLSelectElement;
+        fireEvent.change(areaFilter,
+          { target: { value: testChildCourse.area.name } });
+        const catalogText = 'The table will be filtered as characters are typed in this course filter field';
+        const catalogFilter = renderResult
+          .getByLabelText(catalogText) as HTMLInputElement;
+        fireEvent.change(catalogFilter,
+          { target: { value: `${testChildCourse.prefix} ${testChildCourse.number}` } });
+        // Open the child course modal to edit the "Same As" value
+        const editButton = await renderResult
+          .findByLabelText(`Edit course information for ${testChildCourse.title}`,
+            { exact: false });
+        fireEvent.click(editButton);
+        await renderResult.findByRole('dialog');
+        const sameAsSelect = renderResult.getByLabelText('Same As', { exact: false }) as HTMLSelectElement;
+        fireEvent.change(
+          sameAsSelect,
+          { target: { value: testParentCourse.id } }
+        );
+        const submitButton = renderResult.getByText('Submit');
+        fireEvent.click(submitButton);
+        await wait(() => !renderResult.queryByText('required fields', { exact: false }));
```

Comment:
The tests have been failing frequently for me locally. I think it's an issue where the relationship isn't actually being saved before navigating away from the page. I'd suggest using `waitForElementToBeRemoved()` here instead of `wait()`, and to check for the modal by its role instead.

Reply:
Thanks for pointing this out, Jonathan. I updated both the Schedule and Multi Year Plan tests to use `waitForElementToBeRemoved` instead in commit [18c4db7](https://github.com/seas-computing/course-planner/pull/652/commits/18c4db7647edc1d66c9425105001dd4afa479d77).